text/0018-publish-update.md

```diff
@@ -0,0 +1,180 @@
+
+# Motivation
+[motivation]: #motivation
+
+Right now Qri is a little _too_ "public". Any time I add anything to Qri and go online, I have no tools for saying "I don't want people downloading this dataset", or "this dataset isn't ready for public consumption". This is creating "use paralysis", where I don't feel as comfortable as I should "practicing" on Qri, because any and everything I save is visible to the world. If I make a mistake, I have no clear way of telling another user "hey don't grab these updates!".
+
+Speaking of updates, I don't have any clear mechanism for re-fetching datasets that's been updated since I "added" it. We need a quick and easy way to syncronize my remote copy of a dataset with the lastest version after it's been published.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This RFC introduces two new concepts to Qri that will come with new commands and, will modify some existing behaviors. Publish and Update will work together to allow users to explicitly opt-in to sync points
+
+Because this process will be run frequently, I'd like to make the common use patterns as simple as possible, to the point that `qri save` should come with a `publish` flag that combines saving an update & publishing it into one step.
+
+## Publish
+All datasets start as "unlisted", and the user must `publish` them to make visible to other users. Peers calling `list_datasets` now only see datasets you've published, at version that's been most-recently published. If I create a dataset, then publish, and save changes to that dataset, I must re-publish the new changes for others to see the update.
+
+Whenever I choose, I may "unpublish" a dataset, which will remove it entirely from the list returned by `list_datasets`. It is true that others may still have a copy of my dataset. At present there's not much I can do about it.
+
+#### Publish will integrate with Registries
+Running `publish` with a configured registry should POST a dataset _summary_ to the registry. We haven't defined dataset previews, summaries, and documents yet (a story for another RFC), but once it's done publish will both list the dataset for p2p discovery and push the dataset to the registry.
+
+## Update
+Update brings your data to the latest known version of a dataset. Update comes in two flavors:
+* Updating Someone else's dataset
+* Updating A Dataset You Own
+Both types of update will be accessed via the same command like and API interfaces, they only differ based on the dataset reference in question.
+
+#### Updating a peer's dataset
+Running update on a peer dataset checks to see if the peer is online and calls `get_dataset`, if the version is different, it'll grab that dataset. If the peer in question is offline we can't check for updates. Once publish pushes to the registry we can check the registry for updates.
```

Comment:
Can I `qri update` on a dataset that I haven't added to my own repo yet? If so does that run `qri add` under the hood first? (Note: `add` might change names in the near future, to `fork` or `clone` or something)

Reply:
I think `qri update` on a dataset you don't have should be an error for now. updating something you don't have feels weird, but could see it coming in handy once we start to deal with doling out permissions like "dustmop can run update on this dataset". There's a whole nightmare there waiting for us 😄

Reply:
"Updating A Dataset In Your Namespace"?